### PR TITLE
TEST Coderabbit.ai

### PR DESCRIPTION
**Pull request by swheaton**
Duplicate of https://github.com/voxel51/fiftyone/pull/3985
To test coderabbit.ai ... let's see if it works and if so what it comes up with.

DO NOT MERGE ME!!

## Summary by CodeRabbit

- **New Features**
	- Introduced support for 3D visualization across the application, including new components and updates for rendering 3D shapes, scenes, and assets.
	- Added a new `three_d` media type in GraphQL schema and state management for enhanced 3D data handling.
	- Updated documentation to include guides and examples for working with 3D datasets and visualization features.
- **Refactor**
	- Refactored configuration setup in `vite.config.ts` to allow dynamic importing and more flexible configuration loading.
	- Updated import paths for icons and components to use `@mui/icons-material`.
	- Renamed and updated components to support the new 3D visualization features, including changes in conditional rendering logic and state management.
- **Bug Fixes**
	- Fixed end-to-end tests to correctly query and handle the loading status for 3D canvas elements.
- **Chores**
	- Added the installation of `typing_extensions` to the build documentation workflow for improved type checking.

### CODE REVIEW

File: app/packages/looker-3d/src/action-bar/ToggleBackground.tsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        }}
+        onClick={(e) => {
+          setIsBackgrounOn((prev) => !prev);
+          e.stopPropagation();
+          e.preventDefault();
+        }}
+      />
+    </ActionItem>
+  );
+};
```
[the remark this answers is not in the document]
Comment (ritch) [reply app/packages/looker-3d/src/action-bar/ToggleBackground.tsx:28]: Must fix

File: app/packages/state/src/recoil/groups.ts [hunk 2]
```diff
@@ -226,6 +232,26 @@ export const hasGroupSlices = selector<boolean>({
...
+
+export const hasFo3dSlice = selector<boolean>({
+  key: "hasFo3dSlice",
+  get: ({ get }) => {
+    return (
+      get(groupMediaTypesSet).has("three_d") ||
+      get(groupMediaTypesSet).has("3d")
+    );
+  },
+});
```
[the remark this answers is not in the document]
Comment (ritch) [reply app/packages/state/src/recoil/groups.ts:253]: We agreed to “3D” as the media type. Ensure this is followed for external facing name’s especially media type

File: docs/source/user_guide/app.rst [hunk 3]
```diff
@@ -1180,10 +1180,21 @@ Using the 3D visualizer
...
 
-.. image:: /images/app/app-3d-visualizer.gif
+.. note::
+
+    Deprecation notice:
+
+    The `point-cloud` media type has been deprecated in favor of the
+    `3d` media type. While we'll keep supporting the `point-cloud` media type
+    for backward compatibility, we recommend using the `3d` media type for new
+    datasets.
```
[the remark this answers is not in the document]
Comment (ritch) [reply docs/source/user_guide/app.rst:1195]: Did you add a notice to the groups doc? This is the main place where point cloud is mentioned.

File: app/packages/state/src/recoil/groups.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (sashankaryal) [reply app/packages/state/src/recoil/groups.ts:253]: "3d" is the external facing media type

File: fiftyone/__public__.py [hunk 4]
```diff
@@ -157,6 +157,33 @@
...
+    MeshDepthMaterial,
+    MeshLambertMaterial,
+    MeshPhongMaterial,
+    PointcloudMaterial,
+    Scene,
+    SceneBackground,
+    Euler,
+    Quaternion,
+    Vector3,
+)
```
[the remark this answers is not in the document]
Comment (swheaton) [reply fiftyone/__public__.py:186]: Yes create unit tests for the new `threed` python module located at `fiftyone/core/threed`.
Create one test file per module, each one called `tests/unittests/threed/MODULE_tests.py`, where MODULE is the name of the python file like `object_3d.py`. For example, there are two test files already existing called `tests/unittests/threed/object_3d_tests.py` and `tests/unittests/threed/scene_3d_tests.py`, which correspond to modules `fiftyone/core/threed/object_3d.py` and `fiftyone/core/threed/scene_3d.py`, respectively.

Also please give a shot at documentation as well, let's see what you come up with.

[the remark this answers is not in the document]
Comment (swheaton) [reply fiftyone/__public__.py:186]: Please create the proposed unit test files with appropriate unit tests

[the remark this answers is not in the document]
Comment (swheaton) [reply fiftyone/__public__.py:186]: I want you to generate the full unit test files for these modules, not examples.

> Would you like me to help by adding some initial documentation or unit tests for these new entities?

**Pull request closed without merging**